scripts/validate_docs_snippet_line_numbers.py

```diff
@@ -1,205 +1,189 @@
+import enum
 import glob
-import logging
-import os
+import pprint
 import re
-from collections import defaultdict
 from dataclasses import dataclass
-from typing import Dict, List, Tuple
+from typing import List, Optional, Tuple
 
-logger = logging.getLogger(__name__)
-logger.addHandler(logging.StreamHandler())
-logger.setLevel(logging.INFO)
+# This should be reduced as snippets are added/fixed
+VIOLATION_THRESHOLD = 523
 
+r = re.compile(r"```\w+ file=(.+)")
 
-OPENING_TAG: str = "<snippet>"
-CLOSING_TAG: str = "</snippet>"
-DOCUSAURUS_REGEX: re.Pattern = re.compile(r"```python file=(.+)#L(\d*)(?:-L)?(\d*)")
 
+class Status(enum.Enum):
+    MISSING_BOTH = 0
+    MISSING_OPENING = 1
+    MISSING_CLOSING = 2
+    COMPLETE = 3
```

Comment:
This new script is much more comprehensive - rather than saying this snippet is valid or invalid, it now is able to identify what kind of violation is present.

There are 3 possible violations - the complete absence of surrounding tags, a missing opening tag, or a missing closing tag. By enforcing these numbers, we have the ability to more accurately track the state of our snippets.